### PR TITLE
[CARBONDATA-3235] Fix Rename-Fail & Datamap-creation-Fail

### PR DESCRIPTION
**Pull request by NamanRastogi**
Fixed 2 negative scenarios:

### 1. Alter Table Rename Table Fail
**Problem:** When tabe rename is success in hive, for failed in carbon data store, it would throw exception, but would not go back and undo rename in hive.

**Solution:** A flag to keep check if hive rename has already executed, and of the code breaks after hive rename is done, go back and undo the hive rename.

### 2. Create-Preagregate-Datamap Fail
**Problem:** When (preaggregate) datamap schema is written, but table updation is failed
* call CarbonDropDataMapCommand.processMetadata()
* call dropDataMapFromSystemFolder() -> this is  supposed to delete the folder on disk, but doesnt as the datamap is not yet updated in table, and throws NoSuchDataMapException

**Solution:** Call CarbonDropTableCommand.run() instead of CarbonDropTableCommand.processDatamap(). As CarbonDropTableCommand.processData() deletes actual folders from disk.
       
<br><br>
 - [x] Any interfaces changed?   --->   No
 - [ ] Any backward compatibility impacted?
 - [x] Document update required?   --->   No
 - [ ] Testing done 
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1822/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10080/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2031/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/CarbonAlterTableRenameCommand.scala [hunk 1]
```diff
@@ -165,15 +167,22 @@ private[sql] case class CarbonAlterTableRenameCommand(
       case e: ConcurrentOperationException =>
         throw e
       case e: Exception =>
+        if (hiveRenameSuccess) {
+          sparkSession.sessionState.catalog.asInstanceOf[CarbonSessionCatalog].alterTableRename(
```
Comment (qiuchenjian) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/CarbonAlterTableRenameCommand.scala]: If rename to old table name is fail, how should this scene be handled?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1963/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10215/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2236/

**Comment by kumarvishal09** [comment]
@NamanRastogi Please fixed the build failure

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/CarbonAlterTableRenameCommand.scala [hunk 2]
```diff
@@ -165,15 +167,22 @@ private[sql] case class CarbonAlterTableRenameCommand(
...
+            TableIdentifier(newTableName, Some(oldDatabaseName)),
+            TableIdentifier(oldTableName, Some(oldDatabaseName)),
```
Comment (kevinjmh) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/CarbonAlterTableRenameCommand.scala]: Can we reuse `oldTableIdentifier` and `newTableIdentifier` in `alterTableRenameModel` ?

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/CarbonAlterTableRenameCommand.scala (hunk 1 above)
Comment (kevinjmh) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/CarbonAlterTableRenameCommand.scala]: This is a revert operation when L139-142(hive update) is success but L145-146(carbon meta store update) is failed

**Comment by Indhumathi27** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2184/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2400/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10440/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2190/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2406/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10446/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2200/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2416/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10456/

**Comment by kunal642** [comment]
LGTM

**Comment by kumarvishal09** [comment]
LGTM

**Pull request closed without merging**